core/test/mpi/base/rank_mapping.cpp

```diff
@@ -0,0 +1,133 @@
+/*******************************<GINKGO LICENSE>******************************
+Copyright (c) 2017-2022, the Ginkgo authors
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include <memory>
+
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/core/base/mpi.hpp>
+
+
+#include "core/test/utils.hpp"
+
+
+class MapRankToDevice : public ::testing::Test {
+protected:
+    MapRankToDevice()
+        : comm(MPI_COMM_WORLD),
+          rank(gko::mpi::communicator(comm).rank()),
+          size(gko::mpi::communicator(comm).size()),
+          env({{"MV2_COMM_WORLD_LOCAL_RANK", ""},
+               {"OMPI_COMM_WORLD_LOCAL_RANK", ""},
+               {"MPI_LOCALRANKID", ""},
+               {"SLURM_LOCALID", ""}})
```

Comment:
does changing the environment after MPI_Init during the program has effect?

Reply:
It might. In each test I set the env variable explicitly before I call the mapping function. So if the env var is changed in between that then the test might fail. But I think for our containers that shouldn't be an issue.
Also, I reset the environment to what it was at the time `SetUp` is called, so it won't reset to the changes made in between.